torch/distributed/rpc.py

```diff
@@ -61,11 +74,18 @@ def _init_rpc(name, backend=RpcBackend.PROCESS_GROUP):
     if _agent:
         raise RuntimeError("RPC is already initialized")
 
-    if backend == RpcBackend.PROCESS_GROUP:
+    if rpc_backend == RpcBackend.PROCESS_GROUP:
         from .distributed_c10d import _get_default_group
+
         group = _get_default_group()
         # TODO: add try-except and destroy _agent in all processes if any fails.
-        _agent = ProcessGroupAgent(name, group)
+        _agent = ProcessGroupAgent(self_name, group)
```

Comment:
Do we want to check and throws an error/warn if `self_rank != group.rank()` here?